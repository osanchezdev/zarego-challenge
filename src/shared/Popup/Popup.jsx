import React from 'react';
import { oneOfType, node, arrayOf, bool, func } from 'prop-types';
import styles from './Popup.module.css';

const Popup = ({ children, show, handleClose }) => (
  <div
    role="dialog"
    className={`${styles.popup__overlay} ${
      show ? styles.popup__overlay_show : ''
    }`}
    onClick={handleClose}
  >
    <div
      aria-hidden
      className={styles.popup__wrapper}
      onClick={(e) => e.stopPropagation()}
    >
      {children}
    </div>
  </div>
);

Popup.propTypes = {
  children: oneOfType([arrayOf(node), node]).isRequired,
  show: bool,
  handleClose: func.isRequired,
};

Popup.defaultProps = {
  show: false,
};

export default Popup;
