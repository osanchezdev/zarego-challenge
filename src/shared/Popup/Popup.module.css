.popup__overlay {
  cursor: pointer;
  opacity: 0;
  visibility: hidden;
  position: fixed;
  top: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  animation: fadeOut 300ms ease-out;
}

.popup__overlay_show {
  opacity: 1;
  visibility: visible;
  animation: fadeIn 300ms ease-in;
}

.popup__wrapper {
  cursor: default;
  background: #fff;
  border-radius: 5px;
  box-shadow: 0 6px 30px rgba(0, 0, 0, 0.16);
}

.popup__content {
  margin: 0;
  padding: 0;
}

@keyframes fadeIn {
  0% {
    visibility: visible;
    opacity: 0;
  }
  100% {
    visibility: visible;
    opacity: 1;
  }
}

@keyframes fadeOut {
  0% {
    visibility: visible;
    opacity: 1;
  }
  80% {
    visibility: visible;
    opacity: 0;
  }
  100% {
    visibility: hidden;
    opacity: 0;
  }
}
