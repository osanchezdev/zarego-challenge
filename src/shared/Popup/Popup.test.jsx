import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Popup from './Popup';

describe('Popup component', () => {
  it('should render component', () => {
    const tree = renderer
      .create(
        <Popup handleClose={() => {}}>
          <span>children</span>
        </Popup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render children component', () => {
    const childrenText = 'childrenComponent';
    render(
      <Popup handleClose={() => {}}>
        <div>{childrenText}</div>
      </Popup>
    );
    const children = screen.getByText(childrenText);
    expect(children).toBeInTheDocument();
  });

  it('should execute handleClose when overlay is clicked', () => {
    const handleClose = jest.fn();
    render(
      <Popup handleClose={handleClose}>
        <div>children content</div>
      </Popup>
    );
    const overlay = screen.getByRole('dialog');
    fireEvent.click(overlay);

    expect(handleClose).toHaveBeenCalledTimes(1);
  });
});
