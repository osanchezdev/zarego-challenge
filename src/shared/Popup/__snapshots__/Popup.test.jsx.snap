// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup component should render component 1`] = `
<div
  className="popup__overlay "
  onClick={[Function]}
  role="dialog"
>
  <div
    aria-hidden={true}
    className="popup__wrapper"
    onClick={[Function]}
  >
    <span>
      children
    </span>
  </div>
</div>
`;
